src/Analyzers/CSharp/Tests/SimplifyLinqExpression/CSharpSimplifyLinqExpressionFixAllTests.cs

```diff
@@ -0,0 +1,145 @@
+    }
+}",
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task FixAllInDocumentExplicitCall()
+        {
+
+            var testCode = @"
+using System;
+using System.Linq;
+using System.Collections.Generic;
+
+class C
+{
+    static void M()
+    {
+        IEnumerable<string> test = new List<string> { ""hello"", ""world"", ""!"" };
+        var test1 = [|Enumerable.Where(test, x => x.Equals(""!"")).Any()|];
+        var test2 = [|Enumerable.Where(test, x => x.Equals(""!"")).SingleOrDefault()|];
+        var test3 = [|Enumerable.Where(test, x => x.Equals(""!"")).Last()|];
+        var test4 = [|Enumerable.Where(test, x => x.Equals(""!"")).Count()|];
+        var test5 = [|Enumerable.Where(test, x => x.Equals(""!"")).FirstOrDefault()|];
+    }
+}";
+            var fixedCode = @"
+using System;
+using System.Linq;
+using System.Collections.Generic;
+
+class C
+{
+    static void M()
+    {
+        IEnumerable<string> test = new List<string> { ""hello"", ""world"", ""!"" };
+        var test1 = Enumerable.Any(test, x => x.Equals(""!""));
+        var test2 = Enumerable.SingleOrDefault(test, x => x.Equals(""!""));
+        var test3 = Enumerable.Last(test, x => x.Equals(""!""));
+        var test4 = Enumerable.Count(test, x => x.Equals(""!""));
+        var test5 = Enumerable.FirstOrDefault(test, x => x.Equals(""!""));
+    }
+}";
+            await VerifyCS.VerifyCodeFixAsync(testCode, fixedCode);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineDeclaration)]
+        public async Task NestedInDocument()
+        {
+
+            await new VerifyCS.Test
+            {
+                TestCode = @"
+using System;
+using System.Linq;
+using System.Collections.Generic;
+
+class C
+{
+    static void M()
+    {
+        var test = new List<string> { ""hello"", ""world"", ""!"" };
+        var test1 = [|test.Where(x => x.Equals('!')).Any()|];
+        var test2 = [|test.Where(x => x.Equals('!')).SingleOrDefault()|];
+        var test3 = [|test.Where(x => x.Equals('!')).Last()|];
+        var test4 = test.Where(x => x.Equals('!')).Count();
+        var test5 = [|test.Where(a => [|a.Where(s => s.Equals(""hello"")).FirstOrDefault()|].Equals(""hello"")).FirstOrDefault()|];
+    }
+}",
+                FixedCode = @"
+using System;
+using System.Linq;
+using System.Collections.Generic;
+
+class C
+{
+    static void M()
+    {
+        var test = new List<string> { ""hello"", ""world"", ""!"" };
+        var test1 = test.Any(x => x.Equals('!'));
+        var test2 = test.SingleOrDefault(x => x.Equals('!'));
+        var test3 = test.Last(x => x.Equals('!'));
+        var test4 = test.Where(x => x.Equals('!')).Count();
+        var test5 = test.FirstOrDefault(a => a.FirstOrDefault(s => s.Equals(""hello"")).Equals(""hello""));
+    }
+}",
+            }.RunAsync();
+        }
```

Comment:
can we have tests that validate behavior on queries (it will at lesat verify we don't crash).